miqi/agent/skills.py

```
@@ -197,6 +202,20 @@ def build_skills_summary(self) -> str:
         def escape_xml(s: str) -> str:
             return s.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
 
+        def truncate_description(desc: str) -> str:
+            """Trim to description_max_chars at a sentence boundary."""
+            if description_max_chars is None or len(desc) <= description_max_chars:
+                return desc
+            cut = desc[:description_max_chars]
+            best = -1
+            for sep in (". ", "。", "！", "! ", "？", "? "):
```

Ruff (0.16.1)

[warning] String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)